index.js

```diff
@@ -42,17 +98,45 @@ const redisStore = (...args) => {
         redisCache.get(key, handleResponse(cb, { parse: true }));
       })
     ),
-    del: (key, options, cb) => (
+    mget: (...args) => (
       new Promise((resolve, reject) => {
-        if (typeof options === 'function') {
-          cb = options;
+        let cb;
+        let options = {};
+
+        if (typeof args[args.length - 1] === 'function') {
+          cb = args.pop();
+        }
+
+        if (args[args.length - 1] instanceof Object && args[args.length - 1].constructor === Object) {
+          options = args.pop();
         }
 
         if (!cb) {
           cb = (err, result) => (err ? reject(err) : resolve(result));
         }
-  
-        redisCache.del(key, handleResponse(cb));
+
+        redisCache.mget.apply(redisCache, [...args, handleResponse(cb, { parse: true })]);
+      })
+    ),
+    del: (...args) => (
+      new Promise((resolve, reject) => {
+        let cb;
+        let options = {};
+
+        if (typeof args[args.length - 1] === 'function') {
+          cb = args.pop();
+        }
+
+        if (args[args.length - 1] instanceof Object && args[args.length - 1].constructor === Object) {
+          options = args.pop();
+        }
+
+        if (!cb) {
+          cb = (err, result) => (err ? reject(err) : resolve(result));
+        }
+
+        args.push(handleResponse(cb));
+        redisCache.del.apply(redisCache, args);
```

Comment:
Can you use the same array destructuring notation here as you use at `mget`? `[...args, handleResponse(cb)] `?

Reply:
No prob.